lib/types.nix

```
@@ -373,6 +373,14 @@ rec {
       merge = mergeEqualOption;
     };
 
+    hexStr = mkOptionType {
+      name = "hexStr";
+      description = "hexadecimal value represented as string";
```

I think the hexadecimal is still just a representation of something, presumably bytes.
```suggestion
      description = "bytes represented as a string of hexadecimals";
```

That reminds me, shouldn't we check that it's an even number of hex digits?
Half bytes would be rare. The only case I can think of is short 12-bit hex color codes of three hex digits, but that should just be a distinct type (and probably a type that also allows 24-bit color codes, and a `#`).

```suggestion
      description = "string of hexadecimal bytes";
```

Maybe that's a bit vague. For a more syntax oriented description:

```suggestion
      description = "string of hexadecimal pairs";
```

Or if I'm wrong about half bytes:

```suggestion
      description = "string of hexadecimal digits";
```
```suggestion
      description = "string of hexadecimals";
```

---

In line with https://github.com/NixOS/nixpkgs/pull/301039#discussion_r1575343406:

```suggestion
      description = "string of hexadecimal digits";
```